### PR TITLE
Start work on adding API

### PR DESCRIPTION
**Pull request by numselli**

### CODE REVIEW

File: site/API/routes/oauth/user.mjs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            },
+            200: {
+                type: 'object',
+                properties: {
+                    id: { type: 'string' },
+                    username: { type: 'string' },
+                    avatar: { type: 'string' },
+                    discriminator: { type: 'string' },
+                    locale: { type: 'string' },
+                    avatarURL: { type: 'string' }
```
Comment (DEVTomatoCake) [inline site/API/routes/oauth/user.mjs:21]: No point in storing `avatarURL` as it just duplicated id + avatar

File: site/API/routes/oauth/callback.mjs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            redirect_to: { type: 'string' }
+        },
+        response: {
+        }
+    },
+    handler: async (request, reply) => {
+        const { state, code, redirect_to } = request.query;
+    
+        const tokens = await oauthClient.exchangeCode({
+            code: code,
```
Comment (DEVTomatoCake) [inline site/API/routes/oauth/callback.mjs:27]:
```suggestion
            code,
```

File: site/utils/oauth.mjs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+import {Client} from 'oceanic.js'
```
Comment (DEVTomatoCake) [inline site/utils/oauth.mjs:2]: OAuth should be done with something internal (I use https://pastebin.com/CsgakMjL), no reason to use a bloated external package

**Review by DEVTomatoCake — CHANGES_REQUESTED** [review]

File: site/utils/oauth.mjs (hunk 3 above)
Comment (numselli) [reply site/utils/oauth.mjs:2]: The code you linked does not have any rate limit support, oceanic does. with that being said oceanic is bloated for this usecase and i will be looking into a better way to implement oauth.

**Comment by DEVTomatoCake** [comment]
Superseeded by #2

**Pull request closed without merging**